### DISCUSSION TITLE
How to debug - GCP Pubsub Publish Action from Rule Failed

### DISCUSSION THREAD

**Discussion by tettk** [emqx/emqx#10332] [Q&A]
Hi guys,

I just started trying this out and I have a resource and a rule setup to forward incoming mqtt message to GCP Pubsub Topic. The messages seems to be coming in but the forward action to GCP Pubsub seems to be failing. 

Can someone please point me to a documentation or any notes around troubleshooting? If I can at least see what the error message is on those failures, it would be a very good start!

I'm familiar with MQTT and GCP Pubsub, just new to EMQX.

Thank you all!

**Comment by kjellwinblad** [comment]
Hi! First of all you can check that the GCP Pubsub bridge is working. If you go to bridges in the EMQX dashboard the status should be connected. In the bridge settings dialog there is also a test button. You can press this test button to test the settings and get error information if something goes wrong with the test.

You can also check the EMQX log files. They probably contain the error message corresponding to the failures.

Which version of EMQX are you using?

**Reply by tettk** [reply]
Hi,

Thank you for the quick reply!

I'm on emqx-ee 4.4.16. 
Unfortunately I don't see option for "Bridges" but I do see resources which I have created using the auth key. It is showing available (it also shows the same when I click 'test' when I'm editing this resource.
<img width="1291" alt="Screen Shot 2023-04-05 at 6 47 58 AM" src="https://user-images.githubusercontent.com/1508914/230100204-1a419f4b-2092-4fcf-84c8-c7bdec6fc12b.png">

One question I have, is it recommended that I use version 5.x instead? I just didn't go to 5.x because 4.4.16 was the one that showed up for me when I went through the download pages; so I thought it was the LTS/Stable version.

Please advise. 

Thanks so much.

**Reply by kjellwinblad** [reply]
The 4.4.x have been in production use for longer than 5.x so I think it can be considered as more stable but 5.x. 5.x is a redesign in several aspects and have better scalability for example. Sorry, yes they are called resources in 4.4.x and bridges in 5.x.  Are you still having issues? In that case, have you checked if there is any useful information in the log file?
